erpnext/hr/doctype/salary_slip/salary_slip.py

```diff
@@ -815,6 +819,23 @@ def update_status(self, salary_slip=None):
 				timesheet.set_status()
 				timesheet.save()
 
+	def update_loans(self, status):
+		for loan in self.get_loan_details():
+			doc = frappe.get_doc("Loan", loan.name)
+			if status == "Submit":
+				frappe.db.set_value("Repayment Schedule", loan.repayment_name, "paid", 1)
+				doc.reload()
+				doc.total_amount_paid += loan.total_payment
```

Comment:
We should define a function inside loan.py which will calculate total_payment based on repayment schedule. And that function should be called from here. In this way, it will be reusable function to use it for patching (also pull is better than push).

Reply:
done